git_machete/cmd.py

```diff
@@ -424,6 +425,20 @@ def get_git_dir():
     return git_dir
 
 
+def get_abs_git_dir():
+    global abs_git_dir
+    if not abs_git_dir:
+        try:
+            # git_dir is absolute when we are in a directory of submodule
+            if os.path.isabs(get_git_dir()):
+                abs_git_dir = git_dir
+            else:
+                abs_git_dir = os.path.join(get_root_dir(), get_git_dir())
```

Comment:
Hmmm what about `os.path.abspath` (btw pls see if it works under both Python 2.7 and Python 3.6)? It doesn't depend on possibly undocumented assumptions as the current solution does.